### ISSUE TITLE
Discussion try/except code smell?

### ISSUE THREAD

**Issue by razzeee** [xbmc/generator-kodi-addon#13]
As raised in https://github.com/xbmc/generator-kodi-addon/pull/6#pullrequestreview-17799062

Phil said this code smells as if something is done wrongly.
```py
            try:
                xbmc.log(self.format(record), levels[record.levelno])
            except UnicodeEncodeError:
                xbmc.log(self.format(record).encode('utf-8', 'ignore'), levels[record.levelno])
```

Ronie said
> perhaps this post may be useful to you:
http://forum.kodi.tv/showthread.php?tid=144677
> it explains how to properly handle/avoid unicode errors.
> there's an example log function at the bottom of the post, which i've been using in all my addons for years now, and never had an issue with it.

Let's talk about it, as I feel the try/except is way more pythonic then the one solution named in the forum

**Comment by phil65** [comment]
the issue is not try-except itself, especially since python is more follows EAFP idiom ("Easier to ask for forgiveness than permission")
The issue is that this special case shows that the dev does not know when some object is unicode type or string type.
Best practice is to convert any data coming into application  into unicode so that internally everything is handled as unicode. That way this specific try-except handling is not needed.

**Comment by razzeee** [comment]
Yes, I would prefer that in a perfect world. The problem with that is, that we can't make sure of that via our boilerplate. But what we can do is handling it gracefully when it happens.
